### DISCUSSION TITLE
Get "previous player health" on player_hurt event

### DISCUSSION THREAD

**Discussion by redismx** [saul/demofile#294] [Questions] [answered]
I'm trying to calculate the ADR (average damage per round), and unfortunately, I couldn't find a way to get the health the player had before the damage was done.

The reason why I need it is that the damage done calculation needs to be capped on the remaining health that the player had, i.e if a player has 27 HP and received an AK-47 headshot, the event.dmg_health will be over 100, and I should sum only 27 instead of 100.

I tried to do this:
```ts
    const victim = demoFile.entities.getByUserId(event.userid);
    
    const preDamageHealth = victim.health;
    let damageDone = event.dmg_health;
    const postDamageHealth = event.health;
```

But the `victim.health` at this point is already updated, i.e if the player was killed, it's already 0.

**Comment by saul** [comment, marked as answer]
You can do this by keeping track of every player's last health value when it changes (listen on the entities `change` event). Let me know if you need a hand with the code.

With that said, are you sure you don't want to track `dmg_health`? If you have 1hp and I headshot you, it feels wrong to track that as a 1hp hit.

**Reply by redismx** [reply]
I ended up doing exactly what you suggested right after posting this, tracking the health events, although I did it using the `player_hurt` event. Thanks for offering a hand with the code, I really appreciate your commitment with this project!

As for the calculation, this is what Valve uses and tournament organizers too for calculating ADR. I wanted to do this manually instead of relying on the `round_end` -> player.entities[i].damage_dealt (something like that, forgot know the variable name), because a player can disconnect and I didn't want to lose track of its damage—and, for curiosity, I compared the value I calculated manually and this one from the `player.entities`, and they're the same!

**Reply by freakazzz** [reply]
> u

hello! would be grateful if you can help me with the code for the same question.
